revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+        temp_start_date: The start date of this time period.
+        temp_end_date: The end date of this time period.
+        payment_service_fee_rate: A float from 0-1 representing what percentage of donations that are taken by payment service fees.
+    """
+    def setDonationTableValues(self, filtered_dict, date, proj, temp_start_date, temp_end_date, payment_service_fee_rate):
+        current_dict['filtered_dict']['donations'][date][proj.title]
+        total = sum([donation.amount for donation in Payment.objects.donations(project=proj, start_date=temp_start_date, end_date=temp_end_date)])
+        current_dict['total'] = total
+        current_dict['payment_service_fees'] = total * payment_service_fee_rate
+        current_dict['retained_donations'] = total * (1 - payment_service_fee_rate)
+    
+    """
+    Creates and fills in the 'reinvestments' subdictionary of 'cash_in' and 'cash_out'
+    within the JSON serializable dictionary for a certain time period and project.
+
+    :args:
+        filtered_dict: The JSON serializable dictionary.
+        date: A string represenation of temp_start_date.
+        proj: The project of interest.
+        temp_start_date: The start date of this time period.
+        temp_end_date: The end date of this time period.
+    """
+    def setReinvestmentTableValues(self, filtered_dict, date, proj, temp_start_date, temp_end_date):
+        current_dict['cash_out']['reinvestments'][date][proj.title]
+        total = sum([reinvestment.amount for reinvestment in AdminReinvestment.objects.reinvestments(project=proj, start_date=temp_start_date, end_date=temp_end_date)])
+        current_dict['total'] = total
+
+    """
+    Creates and fills in the 'repayments', 'reinvestments', and 'donations' subdictionary of 'cash_in' and 'cash_out'
+    within the JSON serializable dictionary for a certain time period and project.
+
+    :args:
+        filtered_dict: The JSON serializable dictionary.
+        date: A string represenation of temp_start_date.
+        proj: The project of interest.
+        temp_start_date: The start date of this time period.
+        temp_end_date: The end date of this time period.
+        revolv_earnings_rate: A float from 0-1 representing what percentage of repayments RE-volv takes as earnings.
+        payment_service_fee_rate: A float from 0-1 representing what percentage of donations that are taken by payment service fees.
+    """
+    def setAlTableValues(self, filtered_dict, date, proj, temp_start_date, temp_end_date, revolv_earnings_rate, payment_service_fee_rate):
+        self.setRepaymentTableValues(filtered_dict, date, proj, temp_start_date, temp_end_date, revolv_earnings_rate)
+        self.setDonationTableValues(filtered_dict, date, proj, temp_start_date, temp_end_date, payment_service_fee_rate)
+        self.setReinvestmentTableValues(filtered_dict, date, proj, temp_start_date, temp_end_date)
+
+    """
+    Creates and returns a JsonResponse.
```

Comment:
This function does a bit more than that haha, it's like 80 lines. Could definitely use some more comments